### ISSUE TITLE
Introduce Warehouse page

### ISSUE THREAD

**Issue by RoiJS** [RoiJS/Beelina#136]
In this issue, we will start implementing at least the initial version of Warehouse page. This page is where Administrator manage products information on the warehouse.

Following features must be implemented:
- [ ] Add new warehouse - Skipped
- [ ] Edit existing warehouse - Skipped
- [ ] Deactivate warehouse - Skipped
- [ ] Delete warehouse - Skipped
- [x] Manage products information
- [x] Able to Upload excel file to manage products

We can add more specific points later on.

**Comment by RoiJS** [comment]
I have implemented the warehouse module in a way that the architecture supports multiple warehouses but its not available in the interface for now, we can revisit this feature in the future. 

The page is now called as Warehouse product. This shares similar UI with the Products page in the manager accounts. 

I'm done implementing almost everything with the warehouse module except for the import products feature via excel file. This is the next step.

**Comment by RoiJS** [comment]
Done implementing the endpoint for importing the excel file and extract the product information. This will later be consumed by the front-end. 

When dealing with file uploading in GraphQL hotchocolate, refer to this documentation [https://chillicream.com/docs/hotchocolate/v13/server/files](https://chillicream.com/docs/hotchocolate/v13/server/files).

When testing the endpoint using POSTMAN, refer to this stackoverflow discussions.
* https://stackoverflow.com/questions/63588287/postman-upload-file-and-other-argument-with-graphql
* https://stackoverflow.com/questions/61892306/is-there-any-way-to-upload-files-via-postman-into-a-graphql-api

**Comment by RoiJS** [comment]
To make sure data consistency, I have considered using the Cancellation token to monitor if the server request is orphaned. See here https://www.youtube.com/watch?v=b5dyPJ3zyRg

I have used the Cancellation token when retrieving and updating both panel and warehouse products.

**Comment by RoiJS** [comment]
Now, everything about warehouse module has been implemented. 

This is how import products page looks like:
https://github.com/RoiJS/Beelina/assets/17162501/6497aa43-5e73-412f-ae2e-7b4d2db8e266


We have established a format for the excel file that the user must follow in order to import products successfully. Following fields must be present in the file and the order must be followed:

- Code
- Name
- Additional Quantity
- Price (Optional)
- Unit (Optional)
- Number of Units (Optional)

This import process can not only insert new products but it can also handle updating existing products information. The code is used to determine if the products already exists or not.
